docs/ops/sort/MatrixNMS_8.md

```diff
@@ -0,0 +1,157 @@
+
+**Detailed description**: *MatrixNms* is a multi-phase operation.
+
+First it selects candidate bounding boxes that have higher scores than `score_threshold`. Then for each class at most `nms_top_k` number of candidate boxes are selected. Score of the candidate boxes are then decayed according to Matrix NMS algorithm [Wang et al](https://arxiv.org/abs/2003.10152.pdf).This algorithm is applied independently to each class and each batch element. Boxes of `background_class` are skipped and thus eliminated during the process. Finanly, it thresholds the decayed scores with `post_threshold`, and selects at most `keep_top_k` scoring candidate boxes per batch element.
+
+The Matrix NMS algorithm is described as below:
+1.  Compute `n x n` pairwise IoU matrix `X` for top N bounding boxes. Suppose `n` is the number of candidate boxes, which have been sorted descending by score, thresholded with `score_threshold` and then seleted with `nms_top_k`.
+2.  `triu`: Set the lower triangle and diagonal of `X` to 0.
+3.  Take th column-wise max of `X` to compute a vector `K` of maximun IoU for each candidate box.
+4.  Repeat element value of `K` along axis 1. Suppose this get the matrix `X_cmax`.
+5.  Compute the decay factor: `decay_factor = exp((X_cmax**2 - X**2) * gaussian_sigma)` if `decay_function` is `guassian`, else `decay_factor = (1 - X) / (1 - X_cmax)`.
+6.  Take the column-wise min of `decay_factor`, and element-wise multiply with scores to decay them.
+
+**Attributes**:
+
+* *sort_result*
+
+  * **Description**: *sort_result* specifiy order of output elements.
+  * **Range of values**: `class`, `score`, `none`
+    * *class* - sort selected boxes by class id (ascending) in each batch element.
+    * *score* - sort selected boxes by score (descending) in each batch element.
+    * *none* - do not guarantee the order in each batch element.
+  * **Type**: `string`
+  * **Default value**: `none`
+  * **Required**: *No*
+
+* *output_type*
+
+  * **Description**: the output tensor type
+  * **Range of values**: `i64` or `i32`
+  * **Type**: `string`
+  * **Default value**: `i32`
+  * **Required**: *No*
+
+* *score_threshold*
+
+  * **Description**: minimum score to consider box for the processing.
+  * **Range of values**: a floating-point number
+  * **Type**: `float`
+  * **Default value**: `0`
+  * **Required**: *No*
+
+* *nms_top_k*
+
+  * **Description**: maximum number of boxes to be selected per class.
+  * **Range of values**: an integer starting from `-1`.
+  * **Type**: `int`
+  * **Default value**: `-1` meaning to keep all boxes
+  * **Required**: *No*
+
+* *keep_top_k*
+
+  * **Description**: maximum number of boxes to be selected per batch element.
+  * **Range of values**: an integer starting from `-1`.
+  * **Type**: `int`
+  * **Default value**: `-1` meaning to keep all boxes
+  * **Required**: *No*
+
+* *background_class*
+
+  * **Description**: the background class id.
+  * **Range of values**: an integer from `-1` to the maximum class id
```

Comment:
Are classes counted from 0? If so, then the maximum value is `max_class_id - 1`.

Reply:
It is tricky here. class id starts from 0. And PaddlePaddle uses -1 meaning keep all classes (no background class). The maximum class id should not be -1. It comes from shape of input scores. So maybe we can specify any integer out of [0, num_class] to mean keep all classes.